cookbook/sonoff-basic-light-switch.rst

```diff
@@ -0,0 +1,176 @@
+    **The PCB has mains electricity flowing through it, if you are in any doubt about your capabilities do not attempt to do
+    this.**
+
+If you have a light switch in a brick wall, an option might be to place the Sonoff Basic above the light in the ceiling void,
+and use the cable that ran from the light to the switch as a low voltage cable to connect the Sonoff GPIO.
+
+.. warning::
+
+    If you are going to reuse existing wiring to connect to the GPIO, you must make sure its connected directly to the switch
+    and does not have mains voltage on it from another circuit.
+
+Implementation
+--------------
+
+So having established the pitfalls that you need to overcome lets crack on with making this work.
+
+Please make sure you have read up about :doc:`the Sonoff Basic and how to flash it with ESPHome </devices/sonoff_basic>`.
+As that won't be covered here.Also make sure you know your way around a soldering iron and can find the relevant information
+about the location of the GPIO pins on the Sonoff Basic if you need to.
+
+If you have a Sonoff Basic V1 devices GPOI14 is alredy presented on a pin header on the PCB next to the programing pins.
+On the V2 and V3 PCBs, there is a solder pad underneath the PCB that will let you get at this GPIO.
+
+You have 2 choices when it comes to picking which GPIO to use. GPIO0 or GPIO14. GPIO0 is used by the push button switch on the
+the PCB so you will need to locate the right pin on the switch and solder a wire onto it if you're going to use that one. Whichever
+one you pick, you will also need to use the ground or 0v pin for the other side of the switch. Once you have soldered your wires
+into palce, a handy tip is to add a drop of glue over the wire, a little way away from the solder joint, so give some strain relief
+to the joint.
+
+Now you have a pair of wires from the GPIO and 0v to your retractive switch lets look at the code.
+
+.. code-block:: yaml
+
+    esphome:
+      name: my_ls
+      platform: ESP8266
+      board: esp01_1m
+
+    wifi:
+      ssid: !secret wifi_ssid
+      password: !secret wifi_password
+      fast_connect: true
+
+    logger:
+      level: INFO
+
+    api:
+
+    ota:
+
+    binary_sensor:
+      - platform: gpio
+        pin:
+          number: GPIO14
+          mode: INPUT_PULLUP
+          inverted: True
+        id: button_1
+        on_press:
+          then:
+            - light.toggle: light_1
+
+      - platform: status
+        name: "My LS Status"
+
+    output:
+      - platform: gpio
+        pin: GPIO12
+        id: relay_1
+
+    light:
+      - platform: binary
+        name: "My Light"
+        id: light_1
+        output: relay_1
+
+    status_led:
+      pin:
+        number: GPIO13
+        inverted: yes
+
+In the above code block, there is a *secrets.yaml* file so that you have just one place to change wifi
+details for all your devices.
+
+The logger level INFO is required to overcome a bug that can stop the Sonoff Basic connecting to the wifi.
```

Comment:
```suggestion
```

This has been fixed since the creation of this PR :) https://github.com/esphome/issues/issues/455